subsys/usb/class/bluetooth.c

```diff
@@ -38,7 +38,7 @@ NET_BUF_POOL_DEFINE(tx_pool, CONFIG_BT_HCI_CMD_COUNT, CMD_BUF_SIZE,
 #define BT_L2CAP_MTU 64
 /* Data size needed for ACL buffers */
 #define BT_BUF_ACL_SIZE BT_L2CAP_BUF_SIZE(BT_L2CAP_MTU)
-NET_BUF_POOL_DEFINE(acl_tx_pool, 2, BT_BUF_ACL_SIZE, sizeof(u8_t), NULL);
+NET_BUF_POOL_DEFINE(acl_tx_pool, 4, BT_BUF_ACL_SIZE, sizeof(u8_t), NULL);
```

Comment:
Would it make sense to sync this up with `CONFIG_BT_CTLR_TX_BUFFERS`?

Reply:
Ok, sounds feasible. I will fix it.

Reply:
is increase to 4 or to CONFIG_BT_CTLR_TX_BUFFERS justified?  implicit flow control by use of K_FOREVER should be enough to throttle ACL packet into controller, hence the original double buffer should suffice?

Reply:
@mkarhumaa note that `CONFIG_BT_CTLR_TX_BUFFERS` will not be available in some "exotic" configurations, i.e. when you're using USB to proxy a controller that's behind some physical transport rather than the native controller. So you'll probably want hci_usb to have it's own define something along the lines of:

```
#if defined(CONFIG_BT_CTLR_TX_BUFFERS)
#define ACL_BUF_COUNT CONFIG_BT_CTLR_TX_BUFFERS
#else
#define ACL_BUF_COUNT 4
#endif
```

Reply:
> is increase to 4 or to CONFIG_BT_CTLR_TX_BUFFERS justified? 

@cvinayak that's a fair question. I was just thinking that if one sets `CONFIG_BT_CTLR_TX_BUFFERS` to a considerably higher value, then hci_usb will become a bottleneck if its own pool doesn't grow dynamically.

Reply:
> is increase to 4 or to CONFIG_BT_CTLR_TX_BUFFERS justified? implicit flow control by use of K_FOREVER should be enough to throttle ACL packet into controller, hence the original double buffer should suffice?

This was suggested by Johann F. at #usb channel in Slack. I think increasing acl_tx_pool should make sure that it will not become bottleneck and stall on host side.

Reply:
ok, its the speed of USB+DMA vs the memcpy into controller's tx buffers. I would use a low buffer count, and users can be instructed in the form of comment to increase the buffer count upto the CTLR_TX_BUFFERS if degraded throughput is observed. (lets try to keep low RAM usage in our samples).

Reply:
@cvinayak isn't also USB+DMA vs how fast the packets are sent over air and the buffers consequently freed up in the controller? I'd imagine the memcpy to be negligible there.

Reply:
@cvinayak So you prefer keeping acl_tx_pool size in 2 and just adding comment? To my understanding CONFIG_BT_CTLS_RX_BUFFERS defaults to 7 if BT_HCI_RAW is enabled and 3 if not.

Keeping it in 2 is fine for me.